deltacat/compute/compactor/compaction_session.py

```diff
@@ -194,79 +167,83 @@ def _execute_compaction_round(
+    # read the results from any previously completed compaction round
     round_completion_info = None
     if not rebase_source_partition_locator:
-        logger.info(
-            f"Reading round completion file for compatible "
-            f"primary key index root path: {compatible_primary_key_index_root_path}")
         round_completion_info = rcf.read_round_completion_file(
             compaction_artifact_s3_bucket,
-            source_partition_locator,
-            compatible_primary_key_index_root_path,
+            source_partition_locator
         )
+        if not round_completion_info:
+            logger.info(
+                f"Need rebase source to run initial rebase, otherwise provide round completion file for incremental "
+                f"compaction")
+            return None, None, None
         logger.info(f"Round completion file: {round_completion_info}")
 
-    # read the previous compaction round's hash bucket count, if any
-    old_hash_bucket_count = None
-    if round_completion_info:
-        old_pki_version_locator = round_completion_info\
-            .primary_key_index_version_locator
-        old_hash_bucket_count = old_pki_version_locator\
-            .primary_key_index_version_meta \
-            .hash_bucket_count
-        min_pk_index_pa_bytes = round_completion_info.pk_index_pyarrow_write_result.pyarrow_bytes
-    else:
-        logger.info(f"No prior round info read. Source partition: "
-                    f"{source_partition_locator}. Primary key index locator: "
-                    f"{compatible_primary_key_index_locator}. Rebase source "
-                    f"partition locator: {rebase_source_partition_locator}")
-
-    # use the new hash bucket count if provided, or fall back to old count
-    hash_bucket_count = new_hash_bucket_count \
-        if new_hash_bucket_count is not None \
-        else old_hash_bucket_count
-
     # discover input delta files
+    # For rebase:
+    # Copy the old compacted table to a new destination, plus any new deltas from rebased source
+
+    # For incremental compaction:
+    # Input deltas from two sources: One: new delta; Two: compacted table
+
     high_watermark = round_completion_info.high_watermark \
         if round_completion_info else None
 
+    # Source One: new deltas from uncompacted table for incremental compaction, or deltas from compacted table for rebase
     input_deltas = io.discover_deltas(
         source_partition_locator,
-        high_watermark,
-        last_stream_position_to_compact,
+        high_watermark[source_partition_locator.canonical_string()] if isinstance(high_watermark,
+                                                                                  dict) else high_watermark,
+        last_stream_position_to_compact if not rebase_source_partition_locator else deltacat_storage.get_partition(
+            source_partition_locator.stream_locator,
+            source_partition_locator.partition_values).stream_position,
```

Comment:
Isn't `last_steam_position_to_compact` equal to `old_compacted_locator` stream position in case of a rebase? This may be added as a precondition instead of silently passing old compacted latest stream position?

Reply:
you are right. the confusing part is in case of rebase, source_partition_locator actually points to compacted partition locator.